app/models/member.rb

```diff
@@ -246,6 +246,12 @@ def underage?
     !adult?
   end
 
+  def underage_at?(event_date)
+    return true if birth_date.blank?
+
+    return ((18.years.ago + (event_date - Date.today)) < birth_date)
```

Comment:
I think `birth_date + 18.years < event_date` would read better (if that's still correct (code)).